lib/public/components/Filters/common/FilterModel.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * @license
+ * Copyright CERN and copyright holders of ALICE O2. This software is
+ * distributed under the terms of the GNU General Public License v3 (GPL
+ * Version 3), copied verbatim in the file "COPYING".
+ *
+ * See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+import { Observable } from '/js/src/index.js';
+
+/**
+ * Model storing the state of a given filter
+ */
+export class FilterModel extends Observable {
+    /**
+     * Constructor
+     */
+    constructor() {
+        super();
+
+        this._visualChange$ = new Observable();
+    }
+
+    /**
+     * Reset the filter to its initial state
+     * @return {void}
+     */
+    reset() {
+    }
+
+    /**
+     * States if the filter has been filled with a valid value
+     *
+     * @return {boolean} true if the filter is filled
+     */
+    get isEmpty() {
+        return true;
+    }
+
+    /**
+     * Returns the normalized value of the filter, that can be used as URL parameter
+     *
+     * @return {string|number|object|array|null} the normalized value
+     */
+    get normalized() {
+        return null;
+    }
```

Comment:
I have doubts about this method name:
1. If it is intended as general purpose accessor (getter) of internally stored values (with maybe some not serious adaptation) it might be just getValue (or something similar)
2. If it is intended to return values already suitable to be injected into search query params, it might be `toSeachQueryFormat` (or sth else)

Reply:
I have put this because it's a common function in every filter that return the filter's data in a normalized way. It's not really the value (for example value can be a date, but normalized will be a timestamp). It's not only search query format, because this can be an object, or nested objects... Not sure if I'm clear